internal/txapp/txapp.go

```diff
@@ -38,23 +42,25 @@ func NewTxApp(db DB, engine common.Engine, signer *auth.Ed25519Signer,
 	slices.Sort(resTypes)
 
 	t := &TxApp{
-		Database: db,
-		Engine:   engine,
-		events:   events,
-		log:      log,
+		Database:    db,
+		Engine:      engine,
+		chainParams: chainParams,
+		events:      events,
+		log:         log,
 		mempool: &mempool{
 			accounts:   make(map[string]*types.Account),
-			gasEnabled: GasEnabled,
+			gasEnabled: !chainParams.ConsensusParams.WithoutGasCosts,
 			nodeAddr:   signer.Identity(),
 		},
 		signer:              signer,
-		chainID:             chainID,
-		GasEnabled:          GasEnabled,
+		chainID:             chainParams.ChainID,
+		GasEnabled:          !chainParams.ConsensusParams.WithoutGasCosts,
```

Comment:
Since we now support updating of these params via a fork, we now need a way to read this data dynamically, right? Probably by passing the current block's consensus params as part of the `TxContext`? If a user were to update their `GasEnabled` via a fork, it would not get updated here.

Reply:
That's a sharp observation, although `WithoutGasCosts` is unchangeable presently because the parameter updates pertain only to the `BaseConsensusParams`:

``
type ConsensusParams struct {
	BaseConsensusParams

	// These are unchangeable after genesis. In fact, WithoutNonces probably doesn't even work.
	WithoutNonces   bool `json:"without_nonces"`
	WithoutGasCosts bool `json:"without_gas_costs"`
}
```

I will have another scan through to see if we're failing to reference the active parameters anywhere else.

Which reminds me, `WithoutNonces` should be eliminated.  I think we've removed any app logic based on it, and there was nothing really sensible in the public key transaction signing infrastructure.